packages/platform-browser/src/dom/dom_renderer.ts

```diff
@@ -317,6 +326,15 @@ class DefaultDomRenderer2 implements Renderer2 {
   }
 
   removeChild(_parent: any, oldChild: any): void {
+    if (this.registry && this.registry.has(oldChild as Element)) {
+      this.registry.animate(oldChild, (timeout: ReturnType<typeof setTimeout>) => {
+        clearTimeout(timeout);
+        this.registry?.remove(oldChild);
+        oldChild.remove();
+      });
+      return;
+    }
+    // child was removed
     oldChild.remove();
```

Comment:
We should try to minimize the amount of non-tree-shakable code here (in favor of having more code inside of the registry itself):
```suggestion
    if (!this.registry?.animateRemoval(oldChild, () => oldChild.remove())) {
      // Either animation registry is not present (i.e. a component doesn't use animations)
      // or that element was not in the registry. If that's the case - remove the element here.
      // Otherwise, it'd be removed by the animation logic at the end of the animation.
      oldChild.remove();
    }
```